src/perfrecord.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+  perfrecord.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Nate Rogers <nate.rogers@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <QDebug>
+#include <QFileInfo>
+
+#include "perfrecord.h"
+
+PerfRecord::PerfRecord(QObject* parent)
+    : QObject(parent)
+    , m_PerfRecordProcess(new QProcess(this))
+    , m_perfCommand()
+    , m_outputPath()
+{
+}
+
+PerfRecord::~PerfRecord() = default;
+
+void PerfRecord::record(const QStringList &perfOptions, const QString &outputPath, const QString &exePath, const QStringList &exeOptions)
+{
+    QFileInfo info(exePath);
+    if (!info.exists()) {
+        emit recordingFailed(tr("File '%1' does not exist.").arg(exePath));
+        return;
+    }
+    if (!info.isFile()) {
+        emit recordingFailed(tr("'%1' is not a file.").arg(exePath));
+        return;
+    }
+    if (!info.isReadable()) {
+        emit recordingFailed(tr("File '%1' is not readable.").arg(exePath));
+        return;
+    }
+
+    connect(m_PerfRecordProcess, static_cast<void (QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished),
+            [this] (int exitCode, QProcess::ExitStatus exitStatus) {
+                qDebug() << exitCode << exitStatus;
+
+                if (exitCode == EXIT_SUCCESS && QFileInfo::exists(m_outputPath)) {
+                    emit recordingFinished(m_outputPath);
+                } else {
+                    emit recordingFailed(tr("Failed to record perf data, error code %1.").arg(exitCode));
+                }
+            });
+
+    connect(m_PerfRecordProcess, &QProcess::errorOccurred,
+            [this] (QProcess::ProcessError error) {
+                qWarning() << error << m_PerfRecordProcess->errorString();
+
+                emit recordingFailed(m_PerfRecordProcess->errorString());
+            });
+
+    m_outputPath = outputPath;
+
+    QLatin1String perfBinary("perf");
+    QStringList perfCommand;
```

Comment:
use C++ initializer list for this, and make `m_perfCommand` a QStringList, i.e.:

```
m_perfCommand = {
    QStringLiteral("record"),
    QStringLiteral("-o"),
    m_outputPath,
};
m_perfCommand += perfOptions;
m_perfCommand += exePath;
m_perfCommand += exeOptions;
```